jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/HttpChannel.java

```diff
@@ -500,7 +496,13 @@ public HttpURI getHttpURI()
         }
 
         @Override
-        public String getPath()
+        public Context getContext()
+        {
+            return getServer().getContext();
+        }
+
+        @Override
+        public String getPathInContext()
```

Comment:
The relation between `getContext().getContextPath()` and this method is unclear, and the fact that they're on different classes add to the confusion.

Reply:
I've imprived the javadoc, but in short: `request.getHttpURI.getPath() == request.getContext().getContextPath() + request.getPathInContext()`

Reply:
Okay, that's much clearer now.